text/0000-xor-urls/0000-xor-urls.md

```diff
@@ -0,0 +1,115 @@
+# XOR-URLs (XOR-name based URLs)
+
+- Status: proposed
+- Type: new feature
+- Related components: Safe Browser. Safe App Nodejs / SAFE Client libs.
+- Start Date: (13-11-2018)
+- Discussion: (fill me in with link to RFC discussion - shepherd will complete this)
+- Supersedes: (fill me in with a link to RFC this supersedes - if applicable)
+- Superseded by: (fill me in with a link to RFC this is superseded by - if applicable)
+
+## Summary
+
+It is herein proposed to have a resolver function in the SAFE app client API that allows us to have standardised `safe://`-URLs which are generated based on the XOR address of the content being referenced.
+
+## Conventions
+- The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
+
+## Motivation
+
+Currently the SAFE app client API, and our SAFE browser, support fetching safesites and content stored on the SAFE Network with the DNS/public-names system, i.e. from URLs like `safe://<service name>.<public name>/<path>`, but it’s not possible to fetch data using their address on the network (i.e. the XOR address of any data stored on the network) without publishing it under a public name.
+
+This represents not only a restriction in some trivial use cases, like sharing a file by sharing its location and without publishing it under a public-name, but it also prevents users from being able to store linked-data on the network as these type of data cannot be linked unless it's all published at a public-name.
+
+## Detailed design
+
+### XOR address encoding
+The XOR address shall be encoded in the XOR-URL, [z-base32 encoding](http://philzimmermann.com/docs/human-oriented-base-32-encoding.txt) seems to be a good choice as it is case insensitive, as opposed to other case-sensitive encodings like base58btc or base64url, and it was designed to be easier for human use permuting the alphabet so that the easier characters are the ones that occur more frequently.
+
+### ImmutableData XOR-URL
+XOR-URLs for ImmutableData’s are the simplest ones since they don’t need any additional information to uniquely identify them on the network, as opposed to MutableData’s that also have a type tag. Therefore an ImmutableData XOR-URL can be simply defined as `safe://<encoded ImmutableData XOR addr>`.
```

Comment:
`safe://<encoded ImmutableData XOR addr>`.

Do we not need to include the encoding func as part of the url?

Reply:
Ah, you go into this below 👍

Reply:
right, yes this is the CID that is detailed below, which is actually part of the proposal, I guess if anyone proposes a different encoding it should also be discussed here.